src/devices/Amg88xx/Amg88xx.cs

```diff
@@ -0,0 +1,475 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.I2c;
+using UnitsNet;
+
+namespace Iot.Device.Amg88xx
+{
+    /// <summary>
+    /// Binding for the AMG88xx family of infrared array sensors
+    /// </summary>
+    public class Amg88xx : IDisposable
+    {
+        /// <summary>
+        /// Standard device address
+        /// (AD_SELECT pin is low, c.f. reference specification, pg. 11)
+        /// </summary>
+        public const int DeafultI2cAddress = 0x68;
+
+        /// <summary>
+        /// Alternative device address
+        /// (AD_SELECT pin is high, c.f. reference specification, pg. 11)
+        /// </summary>
+        public const int AlternativeI2cAddress = 0x69;
+
+        /// <summary>
+        /// Number of sensor pixel array columns
+        /// </summary>
+        public const int Columns = 0x8;
+
+        /// <summary>
+        /// Number of sensor pixel array rows
+        /// </summary>
+        public const int Rows = 0x8;
+
+        /// <summary>
+        /// Number of bytes per pixel
+        /// </summary>
+        private const int BytesPerPixel = 2;
+
+        /// <summary>
+        /// Temperature resolution of thermistor (in degrees Celsius)
+        /// </summary>
+        private const double ThermistorTemperatureResolution = 0.0625;
+
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Amg88xx"/> binding.
+        /// </summary>
+        public Amg88xx(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+        }
+
+        #region Infrared sensor
+
+        /// <summary>
+        /// Gets the current thermal image.
+        /// </summary>
+        /// <returns>Array of pixel temperatures [column, row]</returns>
+        public Temperature[,] GetThermalImage()
+        {
+            // the readout process gets triggered by writing to pixel 0 of the sensor w/o any additional data
+            _i2cDevice.WriteByte((byte)Register.T01L);
+
+            Span<byte> buffer = stackalloc byte[Rows * Columns * BytesPerPixel];
+            _i2cDevice.Read(buffer);
+
+            var image = new Temperature[Columns, Rows];
+
+            int idx = 0;
+            for (int r = 0; r < Rows; r++)
+            {
+                for (int c = 0; c < Columns; c++)
+                {
+                    byte tl = buffer[idx++];
+                    byte th = buffer[idx++];
+                    image[c, r] = Amg88xxUtils.ConvertToTemperature(tl, th);
+                }
+            }
+
+            return image;
+        }
+
+        /// <summary>
+        /// Gets the current raw thermal image from the sensor.
+        /// </summary>
+        /// <returns>Thermal image in two's complement representation</returns>
+        public int[,] GetThermalRawImage()
```

Comment:
should this be Uint16/Int16?

Reply:
@krwq I'm uncertain about this. The pixel value is a 12-bit two's complement (I should add the 12-bits information to the document). So this is a signed value. However, it feels semantically imprecise to me using either UInt16 or Int16. The '16' gives a wrong hint, from my point of view. That's the reason why I chose the rather "neutural" int-type.
Am I wrong with this?

Reply:
why does it feel imprecise? Int16 allows you to store 15 bits + 1 sign bit and you got only 12. Also you say before you need 2 bytes per color

Reply:
@krwq Yes, with the coffee of today, I do agree, ;)